File: hs/21.hs
divsum n = foldl (+) 0 $ filter (\m -> mod n m == 0) [1..quot n 2]

solution n = n == divsum(ds) && n /= ds
    where ds = divsum(n)

main = print $ foldl1 (+) $ filter solution [2..10000]

File: clj/21.clj
(defn divsum [n]
  (apply + (filter #(= 0 (mod n %)) (range 1 (+ 1 (/ n 2))))))

(defn solution [n]
  (let [ds (divsum n)]
    (and (= n (divsum ds)) (not= n ds))))

(prn (apply + (filter solution (range 2 10000))))
